### ISSUE TITLE
cannot install custom version of existing lib anymore

### ISSUE THREAD

**Issue by 12rambau** [openforis/sepal#248]
**Describe the bug**
I didn't test it on every lib but it's an issue for sepal_ui. I wanted to work with a custom installation and I get the following error: 

```
PermissionError: [Errno 13] Permission denied: '/usr/local/bin/activate_venv'
```

I checked the permission and as expected owner has everythong and group and other only read and execute: 

```
$ ls -l /usr/local/bin/activate_venv
-rwxr-xr-x 1 root root 225 Oct 21 09:59 /usr/local/bin/activate_venv
```

**To Reproduce**

```
git clone git@github.com:12rambau/sepal_ui.git
pip install -e sepal_ui/
```

**Expected behavior**

It should replace the default one. As a workaround I'll be using a venv but that is way to complicated for normal users.

**Comment by dfguerrerom** [comment]
I thought it was only related to my sepal env but apparently, not... This happens because when you install `sepal_ui`  locally, it first uninstalls the previous version, this will try to remove `bin` scripts, but I think that in someway it cannot find the local `bin` scripts, and then it tries to uninstall `/user/local/bin` scripts which of course is not possible, throwing `PermissionError`.

In short, I could solve the issue by removing my `.local` folder and installing `sepal_ui` again, not the best way to solve but I couldn't do anything different from my side.

I think the cause of the problem is that we are [adding sepal_ui bin path in manually](https://github.com/openforis/sepal/blob/262fe784637cb97d02095b81ac80e8c92e1f6995/modules/geospatial-toolkit/script/init_post.sh#L12
), but this is no longer required since `sepal_ui` will install the scripts as [entry points](https://github.com/12rambau/sepal_ui/blob/089e5d109d194fd825815b625217629f5b0f9675/setup.py#L126).

**Comment by cdanielw** [comment]
You don't have permission to change the system libraries. You should install it as a user library instead, with the `--user` flag:

`pip install -e sepal_ui/ --user`

**Comment by cdanielw** [comment]
You're absolutely right, pip is clever enough. That means that I get the same behavior without the `--user` flag, and am unable to replicate the problem. Installation adds binaries under `~/.local/bin` and even if I remove that directory before uninstalling and/or reinstalling, it works properly. No attempt to write in `/usr/local/bin/` is made.

**Comment by cdanielw** [comment]
You're absolutely right, pip is clever enough. That means that I get the same behavior without the `--user` flag, and am unable to replicate the problem. Installation adds binaries under `~/.local/bin` and even if I remove that directory before uninstalling and/or reinstalling, it works properly. No attempt to write in `/usr/local/bin/` is made. 

Perhaps you have an old version of the library in `.local`, one that tries to delete files under  `/usr/local/bin/` when uninstalling?

**Comment by 12rambau** [comment]
the solution proposed by @dfguerrerom worked for me. I'm trying to update sepal_ui and I tried with numpy which worked as expected. 

I'll close this when sepal_ui become a better neighbours (WIP)

**Comment by dfguerrerom** [comment]
The cause found in this thread, the sepal_ui bin folder added to PATH at sandbox start, was removed in 24dd98b8d. Please reopen if it happens again.
